### ISSUE TITLE
[MuteToggler@jebeaudet.com] Error message about use of toString()

### ISSUE THREAD

**Issue by claudiux** [linuxmint/cinnamon-spices-applets#5102]
### Applet version/Build date

2023-11-15 23:17:18

### Cinnamon version

5.8.4

### Distribution

Mint 21.2

### Graphics hardware *and* driver used

No matters

### Applet name and maintainer

MuteToggler@jebeaudet.com @jebeaudet

### What happened?

An error message about using toString() appears in `~/.xsession-errors`:
```
Gjs-Message: 00:52:19.618: JS LOG: [LookingGlass/info] MuteToggler: Initializing applet
Gjs-Message: 00:52:19.621: JS LOG: [LookingGlass/info] MuteToggler: Initializing settings
Gjs-Message: 00:52:19.626: JS LOG: [LookingGlass/info] Installing settings for MuteToggler@jebeaudet.com
Gjs-Message: 00:52:19.626: JS LOG: [LookingGlass/info] Settings successfully installed for MuteToggler@jebeaudet.com
Window manager warning: "unassigned" found in configuration database is not a valid value for keybinding "MuteToggler@jebeaudet.com"
Gjs-Message: 00:52:19.627: JS LOG: [LookingGlass/info] MuteToggler: Test mixer command 'amixer scontrols'

(cinnamon:2309): Gjs-WARNING **: 00:52:19.654: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to use TextDecoder.
(Note that array.toString() may have been called implicitly.)
0 _init() ["/usr/share/cinnamon/js/misc/fileUtils.js line 211 > Function":50:34]
1 MyApplet() ["/usr/share/cinnamon/js/misc/fileUtils.js line 211 > Function":128:9]
2 main() ["/usr/share/cinnamon/js/misc/fileUtils.js line 211 > Function":132:19]
3 createApplet() ["/usr/share/cinnamon/js/ui/appletManager.js":597:24]
4 addAppletToPanels() ["/usr/share/cinnamon/js/ui/appletManager.js":372:33]
5 finishExtensionLoad() ["/usr/share/cinnamon/js/ui/appletManager.js":98:13]
6 _init/<() ["/usr/share/cinnamon/js/ui/extension.js":275:31]

Gjs-Message: 00:52:19.654: JS LOG: [LookingGlass/info] MuteToggler: Test mixer command 'amixer -D pulse scontrols'
Gjs-Message: 00:52:19.678: JS LOG: [LookingGlass/info] MuteToggler: Use mixer command 'amixer -D pulse'
Gjs-Message: 00:52:19.720: JS LOG: [LookingGlass/info] MuteToggler: Done initializing applet
Gjs-Message: 00:52:19.750: JS LOG: [LookingGlass/info] Loaded applet MuteToggler@jebeaudet.com in 136 ms
```

### Other information

You can use the `to-string.js` library like Radio3.0@claudiux to solve this problem. Also, please create an option in settings allowing users to remove useless messages in  `~/.xsession-errors`.

**Comment by jebeaudet** [comment]
@claudiux Wouldn't it be better to use `TextDecoder` like the log mention?

**Comment by claudiux** [comment]
@jebeaudet Is this really useful, since error messages are not translated?

**Comment by jebeaudet** [comment]
I honestly have no idea, I was just going through the log message and asking against their recommandation.

**Comment by claudiux** [comment]
Do you have a link, please?

**Comment by jebeaudet** [comment]
@claudiux It's in the log you posted

> (cinnamon:2309): Gjs-WARNING **: 00:52:19.654: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to use TextDecoder.

**Comment by claudiux** [comment]
I just made the PR  #5103. I hope you'll find it to your liking.

**Comment by claudiux** [comment]
> @claudiux It's in the log you posted
> 
> > (cinnamon:2309): Gjs-WARNING **: 00:52:19.654: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to use TextDecoder.
Ah yes. I never used TextDecoder; I will try it.

**Comment by jebeaudet** [comment]
`TextDecoder` is not available on 20.3 unfortunately, though it is available on 21.x.

Let's merge your PRs in the meantime, thanks for that btw!
